packages/babel-plugin-transform-async-generator-functions/src/for-await.ts

```diff
@@ -3,29 +3,19 @@ import type { NodePath } from "@babel/traverse";
 
 const buildForAwait = template(`
   async function wrapper() {
-    var ITERATOR_ABRUPT_COMPLETION = false;
-    var ITERATOR_HAD_ERROR_KEY = false;
-    var ITERATOR_ERROR_KEY;
+    var STEP_KEY = {};
     try {
       for (
-        var ITERATOR_KEY = GET_ITERATOR(OBJECT), STEP_KEY;
-        ITERATOR_ABRUPT_COMPLETION = !(STEP_KEY = await ITERATOR_KEY.next()).done;
-        ITERATOR_ABRUPT_COMPLETION = false
+        var ITERATOR_KEY = GET_ITERATOR(OBJECT);
+        !(STEP_KEY = await ITERATOR_KEY.next()).done;
       ) {
       }
-    } catch (err) {
-      ITERATOR_HAD_ERROR_KEY = true;
-      ITERATOR_ERROR_KEY = err;
     } finally {
       try {
-        if (ITERATOR_ABRUPT_COMPLETION && ITERATOR_KEY.return != null) {
+        if (!STEP_KEY.done && ITERATOR_KEY.return != null) {
```

Comment:
This accesses `.done` on the object twice, which is observable if it is a getter.

Reply:
Can we do something like this?

```js
  async function wrapper() {
    try {
      for (
        var NOT_DONE_KEY, ITERATOR_KEY = GET_ITERATOR(OBJECT), STEP_KEY;
        NOT_DONE_KEY = !(STEP_KEY = await ITERATOR_KEY.next()).done;
      ) {
      }
    } finally {
      try {
        if (NOT_DONE_KEY && ITERATOR_KEY.return != null) {
          await ITERATOR_KEY.return();
        }
      } catch (e) {}
    }
  }

```

Reply:
We need to set `NOT_DONE_KEY` to false after the loop ends to avoid executing `return()` when it throws in `next()`.